### ISSUE TITLE
Bracket pair colouriser support

### ISSUE THREAD

**Issue by tauin** [jdinhify/vscode-theme-gruvbox#43]
vscode recently introduced `editor.bracketPairColorization.enabled` as an option (check [this update](https://code.visualstudio.com/updates/v1_59#_extended-theme-customization-syntax))

these can be customised with the settings `editorBracketHighlight.foregroundx` with the trailing x being replace by the numbers 1-6 and `editorBracketHighlight.unexpectedBracket.foreground` for singular brackets not in a pair

**Comment by jdinhify** [comment]
Thanks @tauin 

I'm curious what sort of colours would you use. Obviously they should be in the gruvbox palette (https://github.com/morhetz/gruvbox), but the question is more about the brightness: if you're using dark mode, would you use the same or darker colours for the brackets?

**Comment by Tuuuukka** [comment]
I've been using these for the `Gruvbox Dark Medium` for a while now:

```
    // BRACKETS
    "editorBracketHighlight.foreground1": "#fabd2f",
    "editorBracketHighlight.foreground2": "#d3869b",
    "editorBracketHighlight.foreground3": "#83a598",
    "editorBracketHighlight.foreground4": "#b8bb26",
    "editorBracketHighlight.foreground6": "#8ec07c",
    "editorBracketHighlight.unexpectedBracket.foreground": "#fb4934"
```
![image](https://user-images.githubusercontent.com/10988493/168758404-84b02ccc-9814-4d8e-b6b2-875f292cf8cf.png)

Don't know if they're perfect, but working ok for me (I do mostly JS, TS, and PHP). What @jdinhify said about the brightness is something to consider. If the palette would have even brighter colors than these, I'd use them to kind of highlight the brackets a little and make them stand out from the rest of the code. But that's a personal preference. Using darker colors makes them kinda hard to read for me, but it's also something you might get used to. 

Why I'm used to brighter ones is probably because I used the VSCode default ones for quite some time as I was too lazy to change them 😄

**Comment by shpigunov** [comment]
Here is my proposal. #54 You can read the reasoning behind specific decisions in the PR. I will be happy to answer any questions.

**Comment by jdinhify** [comment]
Merged & deployed #54
